### DISCUSSION TITLE
PKCE Support

### DISCUSSION THREAD

**Discussion by edhenry** [frankie567/httpx-oauth#322] [Q&A]
Hey there! First of all, thank you for the library! This has been incredibly useful and educational for me!

I have a question around the PKCE support that is outlined in the docs [here](https://frankie567.github.io/httpx-oauth/oauth2/#get_access_token). I noticed from the docs that the `code_verifier` is listed as an `Optional[str]` under the `get_access_token` method defined on the `OAuth2AuthorizeCallback`, but I don't see how one could populate this argument. Is there a way to do this or is the argument added for  potential future use?

Apologies if this seems like a simple question. Security isn't my strong suit!

**Comment by frankie567** [comment]
So, it turns out that @king-jam is currently adding proper support to PKCE parameters in #321 😄 It means that we'll soon support to add PKCE parameters start to end.

---

Regarding the values to set, I recommend you to read this documentation from one of my another projects which explains the basics of PKCE: https://docs.fief.dev/going-further/pkce/

You can check out this implementation to know how to generate code verifiers and hashes suitable for PKCE: https://github.com/fief-dev/fief-python/blob/main/fief_client/pkce.py

I'm wondering if this could be a good addition to HTTPX OAuth 🤔  Could be nice to have those helper functions built-in. What do you think?

**Comment by adrien-f** [comment]
Hello and greetings 👋 

Following https://github.com/frankie567/httpx-oauth/pull/321 being merged, the documentation mentions the new PKCE fields but it appears that a version containing them has not been released yet!

Would that be something possible?

Thank you in advance!
